From 9de845e39e6af50f2c87b1bd09f085b5e8780aab Mon Sep 17 00:00:00 2001
From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Wed, 23 Feb 2011 23:03:24 -0600
Subject: [PATCH 58/60] ASoC: ABE DSP: Fix dma_area for ping-pong buffer

ALSA runtime dma_area had an incorrect additional offset for
ping-pong buffer. Some of the addresses defined in ABE DSP driver
which were already defined in ABE HAL are also removed.

Change-Id: Ib6e954dbb486d937cd0a67015a110cb3eba9d153
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
Signed-off-by: Misael Lopez Cruz <misael.lopez@ti.com>
Integrated-by: Jingdong Lu <jingdong.lu@windriver.com>
---
 sound/soc/omap/omap-abe-dsp.c |    3 +--
 sound/soc/omap/omap-abe-dsp.h |    6 ++----
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/sound/soc/omap/omap-abe-dsp.c b/sound/soc/omap/omap-abe-dsp.c
index 6f3f170..8ebc71f 100644
--- a/sound/soc/omap/omap-abe-dsp.c
+++ b/sound/soc/omap/omap-abe-dsp.c
@@ -1896,8 +1896,7 @@ static int abe_ping_pong_init(struct snd_pcm_hw_params *params,
 				PING_PONG_WITH_MCU_IRQ);
 
 	/* Memory mapping for hw params */
-	runtime->dma_area  = abe->io_base + ABE_DMEM_BASE_OFFSET_MPU +
-				ABE_VM_AESS_OFFSET + dst;
+	runtime->dma_area  = abe->io_base + ABE_DMEM_BASE_OFFSET_MPU + dst;
 	runtime->dma_addr  = 0;
 	runtime->dma_bytes = period_size * 2;
 
diff --git a/sound/soc/omap/omap-abe-dsp.h b/sound/soc/omap/omap-abe-dsp.h
index 90d856d..212ef3b 100644
--- a/sound/soc/omap/omap-abe-dsp.h
+++ b/sound/soc/omap/omap-abe-dsp.h
@@ -24,10 +24,8 @@
 #ifndef __OMAP_ABE_DSP_H__
 #define __OMAP_ABE_DSP_H__
 
-#define ABE_DMEM_BASE_OFFSET_MPU	0x80000		/* Ping start offset of DMEM */
-#define ABE_DMEM_BASE_ADDRESS_MPU	0x49080000L	/* Pong start offset of DMEM */
-#define ABE_DMEM_BASE_OFFSET_PING_PONG	0x4000		/* Ping pong buffer DMEM offset */
-#define ABE_VM_AESS_OFFSET		0x400000	/* AESS VM offset for DMEM */
+/* Ping pong buffer DMEM offset */
+#define ABE_DMEM_BASE_OFFSET_PING_PONG	0x4000
 
 void abe_dsp_mcpdm_shutdown(void);
 void abe_dsp_pm_get(void);
-- 
1.7.4.1

